libtiledbsoma/src/soma/managed_query.cc

```diff
@@ -1244,22 +1405,78 @@ ManagedQuery::_extend_and_evolve_schema_with_details(
         enum_values_in_write.assign(
             (ValueType*)data, (ValueType*)data + num_elems);
     }
+    std::unordered_set<ValueType> unique_values_in_write;
+    for (auto enum_value_in_write : enum_values_in_write) {
```

Comment:
C++ style suggests that you should use the undered_set constructor which accepts an input iterator, rather than writing a manual insertion loop

Reply:
For my education, please provide a reference to whichever "C++ style" you're citing. Thank you!

Reply:
(I haven't found a reference by googling, so I'd love to learn from whatever your preferred style reference is.)

Reply:
it follows from two principles:  write less code, and RAII

Reply:
pushed